src/plugins/dashboard/public/dashboard_app/top_nav/add_new_panel/use_get_dashboard_panels.ts

```diff
@@ -0,0 +1,388 @@
+        name: title,
+        icon,
+        onClick: augmentedCreateNewVisType(visTypeAlias, onClickCb),
+        'data-test-subj': `visType-${name}`,
+        description,
+        order: order ?? 0,
+      };
+    },
+    [augmentedCreateNewVisType]
+  );
+
+  const groupUnwrappedEmbeddableFactoriesMap$ = useMemo(
+    () =>
+      defer(() => {
+        return from(
+          Promise.allSettled(
+            Array.from(embeddable.getEmbeddableFactories()).map<
+              Promise<UnwrappedEmbeddableFactory>
+            >(async (factory) => ({
+              factory,
+              isEditable: await factory.isEditable(),
+            }))
+          )
+        ).pipe(
+          map((result) =>
+            result.reduce((acc, cur) => {
+              if (
+                cur.status === 'fulfilled' &&
+                cur.value &&
+                cur.value.isEditable &&
+                !cur.value.factory.isContainerType &&
+                cur.value.factory.canCreateNew() &&
+                cur.value.factory.type !== 'visualization'
+              ) {
+                acc.push(cur.value);
+              }
+
+              return acc;
+            }, [] as UnwrappedEmbeddableFactory[])
+          ),
+          map((factories) => {
+            const _factoryGroupMap: Record<string, FactoryGroup> = {};
+
+            factories.forEach(({ factory }) => {
+              const { grouping } = factory;
+
+              if (grouping) {
+                grouping.forEach((group) => {
+                  if (_factoryGroupMap[group.id]) {
+                    _factoryGroupMap[group.id].factories.push(factory);
+                  } else {
+                    _factoryGroupMap[group.id] = {
+                      id: group.id,
+                      appName: group.getDisplayName
+                        ? group.getDisplayName({ embeddable: dashboardAPI })
+                        : group.id,
+                      icon: group.getIconType?.({ embeddable: dashboardAPI }),
+                      factories: [factory],
+                      order: group.order ?? 0,
+                    };
+                  }
+                });
+              } else {
+                const fallbackGroup = COMMON_EMBEDDABLE_GROUPING.other;
+
+                if (!_factoryGroupMap[fallbackGroup.id]) {
+                  _factoryGroupMap[fallbackGroup.id] = {
+                    id: fallbackGroup.id,
+                    appName: fallbackGroup.getDisplayName
+                      ? fallbackGroup.getDisplayName({ embeddable: dashboardAPI })
+                      : fallbackGroup.id,
+                    icon: fallbackGroup.getIconType?.({ embeddable: dashboardAPI }) || 'empty',
+                    factories: [],
+                    order: fallbackGroup.order ?? 0,
+                  };
+                }
+
+                _factoryGroupMap[fallbackGroup.id].factories.push(factory);
+              }
+            });
+
+            return _factoryGroupMap;
```

Comment:
Interestingly, even with a platinum license, I'm not seeing any factories returned from the `groupUnwrappedEmbeddableFactoriesMap`. Many embeddables have migrated to React embeddables which do not apply here. The remaining class based embeddables are excluded by other conditionals, e.g. Lens `canCreateNew` returns false, Controls are containers, visualize exclusion is hard-coded. 

Unless I'm missing something, we might be able to completely remove this function. But maybe we should wait until all class-based embeddables are removed? @nreese @ThomThomson  what do you think?

Reply:
Great research! I think we should be removing this kind of dead-code as soon as we possibly can.

Reply:
Great research! I think we should be removing this kind of dead-code as soon as we possibly can.

Reply:
@eokoneyo  and I discussed offline and we agreed to remove this `groupUnwrappedEmbeddableFactoriesMap$` function from the code. There are currently only eight embeddables that are returned from the `getEmbeddablesFactories` method. 

1. dashboard - which is a container so it would be filtered out
2. lens - which is filtered out because `canCreateNew` on the factory returns `false`.
3. search - is also filtered out because `canCreateNew` returns `false`.
4. 4. visualization - is hard-coded to be filtered out
5 - 8. controlGroup, optionsList, rangeSlider, timeSlider which all return `false` from the `canCreateNew` method.

Additionally, all of these embeddables are in the process of being converted to React embeddables which are registered with the Add panel flyout using triggers. It is highly unlikely that Lens or Search embeddables will change the return value of their `canCreateNew` methods before the conversion to React embeddables is complete. So I am confident in removing this legacy function.